<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QRegExpWizardWidget</class>
 <widget class="QWidget" name="QRegExpWizardWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>749</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QRegExp Wizard</string>
  </property>
  <property name="sizeGripEnabled" stdset="0">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout">
     <item>
      <widget class="QLabel" name="variableLabel">
       <property name="text">
        <string>&amp;Variable Name:</string>
       </property>
       <property name="buddy">
        <cstring>variableLineEdit</cstring>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="variableLineEdit"/>
     </item>
    </layout>
   </item>
   <item>
    <widget class="Line" name="variableLine">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Pattern Syntax:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="syntaxCombo">
       <property name="toolTip">
        <string>Select the pattern syntax</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Fixed</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>16</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QToolButton" name="undoButton">
       <property name="toolTip">
        <string>&lt;b&gt;Undo last edit&lt;/b&gt;</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="redoButton">
       <property name="toolTip">
        <string>&lt;b&gt;Redo last edit&lt;/b&gt;</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QFrame" name="regexpButtonsFrame">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QToolButton" name="charButton">
        <property name="toolTip">
         <string>&lt;b&gt;Single character of a range (e.g. [abcd])&lt;/b&gt;&lt;p&gt;Select a single character of a range via a specific dialog.&lt;/p&gt;</string>
        </property>
        <property name="whatsThis">
         <string>&lt;b&gt;Single character of a range (e.g. [abcd])&lt;/b&gt;&lt;p&gt;Select a single character of a range via a specific dialog. This dialog will help to edit the range of characters and add some specific conditions.&lt;/p&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="anycharButton">
        <property name="toolTip">
         <string>&lt;b&gt;Any character: '.'&lt;/b&gt;
&lt;p&gt;Select to insert a dot (.) in your regexp.&lt;/p&gt;</string>
        </property>
        <property name="whatsThis">
         <string>&lt;b&gt;Any character: '.'&lt;/b&gt;
&lt;p&gt;Select to insert a dot (.) in your regexp. The dot matches a single character, except line break characters (by default). 
E.g. 'gr.y' matches 'gray', 'grey', 'gr%y', etc. Use the dot sparingly. Often, a character class or negated
character class is faster and more precise.&lt;/p&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="repeatButton">
        <property name="toolTip">
         <string>&lt;b&gt;Repeat contents&lt;/b&gt;
&lt;p&gt;Select a repetition condition via a specific dialog. This dialog will help to specify the allowed range for repetitions.&lt;/p&gt;</string>
        </property>
        <property name="whatsThis">
         <string>&lt;b&gt;Repeat contents&lt;/b&gt;
&lt;p&gt;Select a repetition condition via a specific dialog. This dialog will help to specify the allowed range for repetitions.&lt;/p&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="nonGroupButton">
        <property name="toolTip">
         <string>&lt;b&gt;Non capturing parentheses: (?:)&lt;/b&gt;
&lt;p&gt;Select to insert some non capturing brackets.&lt;/p&gt;</string>
        </property>
        <property name="whatsThis">
         <string>&lt;b&gt;Non capturing parentheses: (?:)&lt;/b&gt;
&lt;p&gt;Select to insert some non capturing brackets. It can be used to apply a regexp quantifier (eg. '?' or '+') to the entire
group of characters inside the brakets. E.g. the regex 'Set(?:Value)?' matches 'Set' or 'SetValue'. The '?:' inside the brakets
means that the content of the match (called the backreference) is not stored for further use.&lt;/p&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="groupButton">
        <property name="toolTip">
         <string>&lt;b&gt;Group: ()&lt;/b&gt;
&lt;p&gt;Select to insert some capturing brackets.&lt;/p&gt;</string>
        </property>
        <property name="whatsThis">
         <string>&lt;b&gt;Group: ()&lt;/b&gt;
&lt;p&gt;Select to insert some capturing brackets. They can be used to apply a regexp quantifier (e.g. '?' or '+') to the entire group of 
characters inside the brakets. E.g. the regex 'Set(Value)?' matches 'Set' or 'SetValue'. Contrary to non-capturing parentheses, 
the backreference matched inside the brakets is stored for further use (i.e. 'Value' in the second example above). 
One can access the backereference with the '\1' expression. &lt;/p&gt;
&lt;p&gt;E.g. '([a-c])x\1x\1' will match 'axaxa', 'bxbxb' and 'cxcxc'.&lt;/p&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="altnButton">
        <property name="toolTip">
         <string>&lt;b&gt;Alternatives: '|'&lt;/b&gt;
&lt;p&gt;Select to insert the alternation symbol '|'. &lt;/p&gt;</string>
        </property>
        <property name="whatsThis">
         <string>&lt;b&gt;Alternatives: '|'&lt;/b&gt;
&lt;p&gt;Select to insert the alternation symbol '|'. The alternation is used to match a single regular expression out of 
several possible regular expressions. E.g. 'cat|dog|mouse|fish' matches words containing the word 'cat', 'dog','mouse' or 'fish'.
Be aware that in the above example, the alternatives refer to whole or part of words. If you want to match exactly the
 words 'cat', 'dog', ... you should express the fact that you only want to match complete words: '\b(cat|dog|mouse|fish)\b'&lt;/p&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="beglineButton">
        <property name="toolTip">
         <string>&lt;b&gt;Begin of line: '^'&lt;/b&gt;
&lt;p&gt;Select to insert the start line character (^).&lt;/p&gt;</string>
        </property>
        <property name="whatsThis">
         <string>&lt;b&gt;Begin of line: '^'&lt;/b&gt;
&lt;p&gt;Select to insert the start line character (^). It is used to find some expressions at the begining of lines.
E.g. '^[A-Z]' match lines starting with a capitalized character. &lt;/p&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="endlineButton">
        <property name="toolTip">
         <string>&lt;b&gt;End of line: '$'&lt;/b&gt;
&lt;p&gt;Select to insert the end of line character ($).&lt;/p&gt;</string>
        </property>
        <property name="whatsThis">
         <string>&lt;b&gt;End of line: '$'&lt;/b&gt;
&lt;p&gt;Select to insert the end of line character ($). It is used to find some expressions at the end of lines.&lt;/p&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="wordboundButton">
        <property name="toolTip">
         <string>&lt;b&gt;Word boundary&lt;/b&gt;
&lt;p&gt;Select to insert the word boudary character (\b).&lt;/p&gt;</string>
        </property>
        <property name="whatsThis">
         <string>&lt;b&gt;Word boundary&lt;/b&gt;
&lt;p&gt;Select to insert the word boudary character (\b). This character is used to express the fact that word 
must begin or end at this position. E.g. '\bcat\b' matches exactly the word 'cat' while 'concatenation' is ignored.&lt;/p&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="nonwordboundButton">
        <property name="toolTip">
         <string>&lt;b&gt;Non word boundary&lt;/b&gt;
&lt;p&gt;Select to insert the word boudary character (\B). \B is the negated version of \b.&lt;/p&gt;</string>
        </property>
        <property name="whatsThis">
         <string>&lt;b&gt;Non word boundary&lt;/b&gt;
&lt;p&gt;Select to insert the word boudary character (\B). \B is the negated version of \b. \B matches at every position where \b 
does not. Effectively, \B matches at any position between two word characters as well as at any position between two non-word characters.&lt;/p&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="poslookaheadButton">
        <property name="toolTip">
         <string>&lt;b&gt;Positive lookahead: (?=&lt;i&gt;regexpr&lt;/i&gt;)&lt;/b&gt;
&lt;p&gt;Select to insert the positive lookhead brackets.&lt;/p&gt;</string>
        </property>
        <property name="whatsThis">
         <string>&lt;b&gt;Positive lookahead: (?=&lt;i&gt;regexpr&lt;/i&gt;)&lt;/b&gt;
&lt;p&gt;Select to insert the positive lookhead brackets. Basically, positive lookhead is used to match a character only if followed by another one.
Writting 'q(?=u)' means that you want to match the 'q' character only if it is followed by 'u'. In this statement 'u' is a trivial 
regexp which may be replaced by a more complex expression; q(?=[abc])' will match a 'q' if followed by either 'a', 'b' or 'c'.&lt;/p&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="neglookaheadButton">
        <property name="toolTip">
         <string>&lt;b&gt;Negative lookahead: (?!&lt;i&gt;regexpr&lt;/i&gt;)&lt;/b&gt;
&lt;p&gt;Select to insert the negative lookhead brackets.&lt;/p&gt;</string>
        </property>
        <property name="whatsThis">
         <string>&lt;b&gt;Negative lookahead: (?!&lt;i&gt;regexpr&lt;/i&gt;)&lt;/b&gt;
&lt;p&gt;Select to insert the negative lookhead brackets. Basically, negative lookhead is used to match a character only if it is not
followed by a another one. Writting 'q(?!u)' means that you want to match 'q' only if it is not followed by 'u'. In this statement, 'u' is a
trivial regexp which may be replaced by a more complex expression; 'q(?![abc])' will match a 'q' if it is followed by anything else than 'a', 'b' or 'c'.&lt;/p&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Expanding</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>356</width>
          <height>24</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="wildcardButtonsFrame">
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QToolButton" name="wildcardCharButton">
        <property name="toolTip">
         <string>&lt;b&gt;Single character of a range (e.g. [abcd])&lt;/b&gt;&lt;p&gt;Select a single character of a range via a specific dialog.&lt;/p&gt;</string>
        </property>
        <property name="whatsThis">
         <string>&lt;b&gt;Single character of a range (e.g. [abcd])&lt;/b&gt;&lt;p&gt;Select a single character of a range via a specific dialog. This dialog will help to edit the range of characters and add some specific conditions.&lt;/p&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="wildcardAnycharButton">
        <property name="toolTip">
         <string>&lt;b&gt;Any character: '.'&lt;/b&gt;
&lt;p&gt;Select to insert a question mark (?) in your regexp.&lt;/p&gt;</string>
        </property>
        <property name="whatsThis">
         <string>&lt;b&gt;Any character: '.'&lt;/b&gt;
&lt;p&gt;Select to insert a question mark (?) in your regexp. The question mark matches a single character. 
E.g. 'gr?y' matches 'gray', 'grey', 'gr%y', etc.&lt;/p&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="wildcardRepeatButton">
        <property name="toolTip">
         <string>&lt;b&gt;Repeat contents&lt;/b&gt;
&lt;p&gt;Inserts a repetition (*) character into the regexp.&lt;/p&gt;</string>
        </property>
        <property name="whatsThis">
         <string>&lt;b&gt;Repeat contents&lt;/b&gt;
&lt;p&gt;Inserts a repetition (*) character into the regexp. That will match zero or more of any character.&lt;/p&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>635</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="w3cButtonsFrame">
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QToolButton" name="w3cCharButton">
        <property name="toolTip">
         <string>&lt;b&gt;Single character of a range (e.g. [abcd])&lt;/b&gt;&lt;p&gt;Select a single character of a range via a specific dialog.&lt;/p&gt;</string>
        </property>
        <property name="whatsThis">
         <string>&lt;b&gt;Single character of a range (e.g. [abcd])&lt;/b&gt;&lt;p&gt;Select a single character of a range via a specific dialog. This dialog will help to edit the range of characters and add some specific conditions.&lt;/p&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="w3cAnycharButton">
        <property name="toolTip">
         <string>&lt;b&gt;Any character: '.'&lt;/b&gt;
&lt;p&gt;Select to insert a dot (.) in your regexp.&lt;/p&gt;</string>
        </property>
        <property name="whatsThis">
         <string>&lt;b&gt;Any character: '.'&lt;/b&gt;
&lt;p&gt;Select to insert a dot (.) in your regexp. The dot matches a single character, except line break characters (by default). 
E.g. 'gr.y' matches 'gray', 'grey', 'gr%y', etc. Use the dot sparingly. Often, a character class or negated
character class is faster and more precise.&lt;/p&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="w3cRepeatButton">
        <property name="toolTip">
         <string>&lt;b&gt;Repeat contents&lt;/b&gt;
&lt;p&gt;Select a repetition condition via a specific dialog. This dialog will help to specify the allowed range for repetitions.&lt;/p&gt;</string>
        </property>
        <property name="whatsThis">
         <string>&lt;b&gt;Repeat contents&lt;/b&gt;
&lt;p&gt;Select a repetition condition via a specific dialog. This dialog will help to specify the allowed range for repetitions.&lt;/p&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="w3cGroupButton">
        <property name="toolTip">
         <string>&lt;b&gt;Group: ()&lt;/b&gt;
&lt;p&gt;Select to insert some capturing brackets.&lt;/p&gt;</string>
        </property>
        <property name="whatsThis">
         <string>&lt;b&gt;Group: ()&lt;/b&gt;
&lt;p&gt;Select to insert some capturing brackets. They can be used to apply a regexp quantifier (e.g. '?' or '+') to the entire group of 
characters inside the brakets. E.g. the regex 'Set(Value)?' matches 'Set' or 'SetValue'.&lt;/p&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="w3cAltnButton">
        <property name="toolTip">
         <string>&lt;b&gt;Alternatives: '|'&lt;/b&gt;
&lt;p&gt;Select to insert the alternation symbol '|'. &lt;/p&gt;</string>
        </property>
        <property name="whatsThis">
         <string>&lt;b&gt;Alternatives: '|'&lt;/b&gt;
&lt;p&gt;Select to insert the alternation symbol '|'. The alternation is used to match a single regular expression out of 
several possible regular expressions. E.g. 'cat|dog|mouse|fish' matches words containing the word 'cat', 'dog','mouse' or 'fish'.&lt;/p&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>573</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QGridLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="textLabel1">
       <property name="text">
        <string>&amp;Regexp:</string>
       </property>
       <property name="buddy">
        <cstring>regexpLineEdit</cstring>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="textLabel2">
       <property name="text">
        <string>&amp;Text:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignTop</set>
       </property>
       <property name="buddy">
        <cstring>textTextEdit</cstring>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="regexpLineEdit"/>
     </item>
     <item row="1" column="1">
      <widget class="QTextEdit" name="textTextEdit"/>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout">
     <item>
      <widget class="QCheckBox" name="caseSensitiveCheckBox">
       <property name="text">
        <string>Case &amp;Sensitive</string>
       </property>
       <property name="shortcut">
        <string>Alt+S</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="minimalCheckBox">
       <property name="text">
        <string>&amp;Minimal</string>
       </property>
       <property name="shortcut">
        <string>Alt+M</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Expanding</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QTableWidget" name="resultTable">
     <property name="editTriggers">
      <set>QAbstractItemView::NoEditTriggers</set>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Close|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="6"/>
 <pixmapfunction>qPixmapFromMimeSource</pixmapfunction>
 <tabstops>
  <tabstop>variableLineEdit</tabstop>
  <tabstop>syntaxCombo</tabstop>
  <tabstop>undoButton</tabstop>
  <tabstop>redoButton</tabstop>
  <tabstop>charButton</tabstop>
  <tabstop>anycharButton</tabstop>
  <tabstop>repeatButton</tabstop>
  <tabstop>nonGroupButton</tabstop>
  <tabstop>groupButton</tabstop>
  <tabstop>altnButton</tabstop>
  <tabstop>beglineButton</tabstop>
  <tabstop>endlineButton</tabstop>
  <tabstop>wordboundButton</tabstop>
  <tabstop>nonwordboundButton</tabstop>
  <tabstop>poslookaheadButton</tabstop>
  <tabstop>neglookaheadButton</tabstop>
  <tabstop>wildcardCharButton</tabstop>
  <tabstop>wildcardAnycharButton</tabstop>
  <tabstop>wildcardRepeatButton</tabstop>
  <tabstop>w3cCharButton</tabstop>
  <tabstop>w3cAnycharButton</tabstop>
  <tabstop>w3cRepeatButton</tabstop>
  <tabstop>w3cGroupButton</tabstop>
  <tabstop>w3cAltnButton</tabstop>
  <tabstop>regexpLineEdit</tabstop>
  <tabstop>textTextEdit</tabstop>
  <tabstop>caseSensitiveCheckBox</tabstop>
  <tabstop>minimalCheckBox</tabstop>
  <tabstop>resultTable</tabstop>
  <tabstop>buttonBox</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>undoButton</sender>
   <signal>clicked()</signal>
   <receiver>regexpLineEdit</receiver>
   <slot>undo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>490</x>
     <y>132</y>
    </hint>
    <hint type="destinationlabel">
     <x>487</x>
     <y>163</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>redoButton</sender>
   <signal>clicked()</signal>
   <receiver>regexpLineEdit</receiver>
   <slot>redo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>526</x>
     <y>132</y>
    </hint>
    <hint type="destinationlabel">
     <x>529</x>
     <y>163</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
